opm/simulators/flow/FlowProblem.hpp

```diff
@@ -340,144 +258,9 @@ class FlowProblem : public GetPropType<TypeTag, Properties::BaseProblem>
     /*!
      * \copydoc FvBaseProblem::finishInit
      */
-    void finishInit()
+    virtual void finishInit()
```

Comment:
why is this made virtual? do we ever call it in a FlowProblem context? if there is an actual reason, you need to mark the methods in FlowProblemBlackoil as override. same question goes for all methods marked virtual (not the pure virtuals, i get that those are for flowproblem calling its subclasses).

Reply:
Thanks. I will try to address this as much as possible.  Some virtual was added to remove the warning of of hiding function of the base class (FlowProblem.) even though we did not call the function in the FlowProblem only called them in the function in the subclass throw `FlowProblemType::endTimeStep();` or something.  Let me know if you have some comment here. 

But the finishInit() function in the FlowProblem can be removed indeed. 

Later on, we can decide whether we want to instantiate FlowProblem (not its subclasses) at all. With its current form,  since we only have blackoil simulators, we can not instantiate FlowProblem.

Reply:
Right. That is a valid reason and none of these are performance critical so being in the vtable doesn't really hurt. I'd add a comment explaining this fact and you still have to mark them override in the subclass or you are just trading one warning for another (with clang, gcc won't warn).

Reply:
For the function `serializeOp`, as a template function, it can not be virtual.